tests/runtest.py

```diff
@@ -764,6 +765,58 @@ def run_tests(host_os,
     if test_env is not None:
         os.environ["__TestEnv"] = test_env
 
+    #=====================================================================================================================================================
+    #
+    # This is a workaround needed to unblock our CI (in particular, Linux/arm and Linux/arm64 jobs) from the following failures appearing almost in every
+    # pull request (but hard to reproduce locally)
+    #
+    #   System.IO.FileLoadException: Could not load file or assembly 'Exceptions.Finalization.XUnitWrapper, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. 
+    #   An operation is not legal in the current state. (Exception from HRESULT: 0x80131509 (COR_E_INVALIDOPERATION))
+    #
+    # COR_E_INVALIDOPERATION comes from System.InvalidOperationException that is thrown during AssemblyLoadContext.ResolveUsingResolvingEvent
+    # when multiple threads attempt to modify an instance of Dictionary (managedAssemblyCache) during Xunit.DependencyContextAssemblyCache.LoadManagedDll call.
+    #
+    # In order to mitigate the failure we built our own xunit.console.dll with ConcurrentDictionary used for managedAssemblyCache and use this instead of
+    # the one pulled from NuGet. The exact code that got built can be found at the following fork of Xunit
+    #  * https://github.com/echesakovMSFT/xunit/tree/UseConcurrentDictionaryInDependencyContextAssemblyCache
+    #
+    # The assembly was built using Microsoft Visual Studio v15.9.0-pre.4.0 Developer Command Prompt using the following commands
+    #  1) git clone https://github.com/echesakovMSFT/xunit.git --branch UseConcurrentDictionaryInDependencyContextAssemblyCache --single-branch
+    #  2) cd xunit
+    #  3) git submodule update --init
+    #  4) powershell .\build.ps1
+    #
+    # Then file "xunit\src\xunit.console\bin\Release\netcoreapp2.0\xunit.console.dll" was archived and uploaded to the clrjit blob storage.
+    #
+    # Ideally, this code should be removed when we find a more robust way of running Xunit tests.
+    #  
+    # References:
+    #  * https://github.com/dotnet/coreclr/issues/20392
+    #  * https://github.com/dotnet/coreclr/issues/20594
+    #  * https://github.com/xunit/xunit/issues/1842 
+    #  * https://github.com/xunit/xunit/pull/1846
+    #
+    #=====================================================================================================================================================
+
+    print("Download and overwrite xunit.console.dll in Core_Root")
+
+    try:
+        import urllib.request
```

Comment:
I would much rather prefer this be wrapped in a if python version < 3 then catching the exception. We will eventually need to remove these python2 isms and this setup makes that work more difficult.

Reply:
I would suggest this import be conditional at the header of the file. After the last from import add something like:

```
# Version specific imports

if sys.version_info < (3,0):
    import urllib.request
else:
    import urllib
```

Reply:
Then you can add the if here as well. I am fine with changing this to a method call is_python3.

```
if sys.version_info < (3,0):
    urlretrieve = urllib.request.urlretrieve
else:
    urlretrieve = urllib.urlretrieve
```